### PR TITLE
Add a context menu to hide "All bookmarks button"

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/6043

Demo:

https://github.com/brave/brave-core/assets/5474642/ec39746a-1d2f-4470-ae5d-8ad6fa16fbd0

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* Automated - unittest & browsertest
* Manual
  * Add a bookmark to "Other bookmarks" node and see if "All bookmarks" button shows up on the bookmarks bar
  * Right click the "All Bookmarks" button and uncheck "Show All bookmarks button"
  * See if the "All bookmarks" button is hidden
  * Vice-versa.

### CODE REVIEW

File: app/brave_command_ids.h [hunk 1]
```diff
@@ -71,18 +71,19 @@
 #define IDC_BRAVE_BOOKMARK_BAR_ALWAYS 56201
 #define IDC_BRAVE_BOOKMARK_BAR_NTP 56202
 #define IDC_BRAVE_BOOKMARK_BAR_NEVER 56203
+#define IDC_TOGGLE_ALL_BOOKMARKS_BUTTON_VISIBILITY 56204
```
Comment (sangwoo108) [inline app/brave_command_ids.h:74]: This is the new one. others are formatted and the values below this item is increased.

**Review by iefremov — APPROVED** [review]
chromium_src lgtm

File: app/brave_command_ids.h (hunk 1 above)
Comment (yrliou) [reply app/brave_command_ids.h:74]: I wonder if we really needs to put it here and change everything below.
It seems we currently does not really do grouping and reserve sections for each in this file. I see there is already a IDC_TOGGLE_VERTICAL_TABS_EXPANDED below.
If we want to start having groups, maybe we should do so by reserving some amount of IDs so we don't change everything after each time?

Comment (fallaciousreasoning) [reply app/brave_command_ids.h:74]: FWIW, I think this could also break shortcuts which use the id to store custom shortcuts

**Review by fallaciousreasoning — APPROVED** [review]
LGTM % nits

File: app/brave_command_ids.h (hunk 1 above)
Comment (sangwoo108) [reply app/brave_command_ids.h:74]: I put this here to group bookmark related commands. But what @fallaciousreasoning says looks critical. Reverted.

Comment (yrliou) [reply app/brave_command_ids.h]: Not really in the scope of this PR, but if you don't mind, please start with 56300 so we could have a range for AI_CHAT group.

File: app/brave_command_ids.h [hunk 2]
```diff
@@ -95,9 +98,13 @@
 
 #define IDC_BRAVE_SEND_TAB_TO_SELF 56225
 
+// Commands for AI chat --------------------------------------------------------
```
Comment (yrliou) [inline app/brave_command_ids.h:101]: nit: just `// AI chat commands.` and ` Bookmark commands` below.

File: browser/ui/views/bookmarks/brave_bookmark_context_menu_unittest.cc [hunk 3]
```diff
@@ -4,19 +4,23 @@
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 #include "brave/browser/ui/views/bookmarks/brave_bookmark_context_menu.h"
+#include "brave/app/brave_command_ids.h"
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply browser/ui/views/bookmarks/brave_bookmark_context_menu_unittest.cc]: Oops, done. Thanks

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**